### DISCUSSION TITLE
v1.0.2

### DISCUSSION THREAD

**Discussion by HathewayWill** [HathewayWill/WRF-MOSIT#9] [Announcements]
Minor fixes to code format.

Added Weblinks, Password Prompt, and Export executable to bashrc prompt.

Release v1.0.2

<hr /><em>This discussion was created from the release <a href='https://github.com/HathewayWill/WRF-MASTER/releases/tag/'>v1.0.2</a>.</em>

**Comment by Debasish-Mahapatra** [comment]
Library Compatibility Tests 
Test 1
01_fortran+c+netcdf_f.f:5: Error: Can't open included file 'netcdf.inc'
gfortran: error: 01_fortran+c+netcdf_f.o: No such file or directory
./WRF_Master_Script.sh: line 8369: ./a.out: No such file or directory
Compatibility Compiler Test 1 Failed

I am getting this error, my I am on Ubuntu 22.04 LTS, with gfortran v 9.5.0. Could you please help?

**Reply by HathewayWill** [reply]
> Library Compatibility Tests 
> Test 1
> 01_fortran+c+netcdf_f.f:5: Error: Can't open included file 'netcdf.inc'
> gfortran: error: 01_fortran+c+netcdf_f.o: No such file or directory
> ./WRF_Master_Script.sh: line 8369: ./a.out: No such file or directory
> Compatibility Compiler Test 1 Failed
> 
> I am getting this error, my I am on Ubuntu 22.04 LTS, with gfortran v 9.5.0. Could you please help?

Sure happy to help.

Did you manually add gfortran v9.5 to your system

**Comment by Debasish-Mahapatra** [comment]
Hii.!

Thanks for the reply.

Yeah I manually installed Gfortran 9.5 and added it to the alternatives and
made it default.

Is this error because of that?


Thanks in advance.
Debasish

**Reply by HathewayWill** [reply]
> Hii.!
> 
> Thanks for the reply.
> 
> Yeah I manually installed Gfortran 9.5 and added it to the alternatives and
> made it default.
> 
> Is this error because of that?
> 
> 
> Thanks in advance.
> Debasish
> 
> On Mon, Jan 23, 2023, 18:10 HathewayWill ***@***.***> wrote:
> 
> > Library Compatibility Tests
> > Test 1
> > 01_fortran+c+netcdf_f.f:5: Error: Can't open included file 'netcdf.inc'
> > gfortran: error: 01_fortran+c+netcdf_f.o: No such file or directory
> > ./WRF_Master_Script.sh: line 8369: ./a.out: No such file or directory
> > Compatibility Compiler Test 1 Failed
> >
> > I am getting this error, my I am on Ubuntu 22.04 LTS, with gfortran v
> > 9.5.0. Could you please help?
> >
> > Sure happy to help.
> >
> > Did you manually add gfortran v9.5 to your system
> >
> > —
> > Reply to this email directly, view it on GitHub
> > <https://github.com/HathewayWill/WRF-MASTER/discussions/9#discussioncomment-4756139>,
> > or unsubscribe
> > <https://github.com/notifications/unsubscribe-auth/A5NGCTDLR5PIL2A2MZ5S5QTWTZ32DANCNFSM6AAAAAAUA3AEJU>
> > .
> > You are receiving this because you commented.Message ID:
> > ***@***.***>
> >
> 


My guess the issue is that you have gfortran v9.5 and gcc and g++ are the default v11 that comes with Ubuntu 22

So the script is trying to use older fortran libraries with newer c libraries and there's a disconnect. 

Two options for you.

1. Download all the GNU compilers for v9.5 and try again. It should work with the old compilers since I programmed the script to do it. 

2. Remove v9.5 and stick with the default v11 GNU compilers. 

The code can do all the installation with either and you won't have to do any commands or line editing for it.

**Reply by Debasish-Mahapatra** [reply]
![Screenshot from 2023-01-24 14-28-51](https://user-images.githubusercontent.com/123363660/214250421-19852d6b-56a0-4d59-98bf-d5cd450a42e8.png)

Please check. I am still getting the same error. It does not work with the old compiler.

**Reply by HathewayWill** [reply]
> ![Screenshot from 2023-01-24 14-28-51](https://user-images.githubusercontent.com/123363660/214250421-19852d6b-56a0-4d59-98bf-d5cd450a42e8.png)
> 
> Please check. I am still getting the same error. It does not work with the old compiler. 

Hmm that is strange. 

Can you type these commands and show me the output. 

gcc --version
gfortran --version
g++ --version

As well as 

which gcc
which gfortran
which g++

**Reply by Debasish-Mahapatra** [reply]
Hii. 

Sorry for the late reply. Please check the requested screenshot.
I have successfully built WRF, WRFPLUS and WRFDA manually with gfortran, gcc and g++ all set to v 9.5. But using the provided script still seems to generate that error.
![Screenshot from 2023-01-27 11-14-54](https://user-images.githubusercontent.com/123363660/215018169-1ad7188b-9f3b-4443-be75-e38d87a369bf.png)

**Reply by HathewayWill** [reply]
> Hii. 
> 
> Sorry for the late reply. Please check the requested screenshot.
> I have successfully built WRF, WRFPLUS and WRFDA manually with gfortran, gcc and g++ all set to v 9.5. But using the provided script still seems to generate that error.
> ![Screenshot from 2023-01-27 11-14-54](https://user-images.githubusercontent.com/123363660/215018169-1ad7188b-9f3b-4443-be75-e38d87a369bf.png)
> 


@Debasish-Mahapatra 

Just noticed something in your screenshot.

You are in a conda environment. (base)

That would be the most likely issue since conda comes with it's own package
manager which my scripts aren't compatible with.

I didn't notice it in the first screenshot because I was looking at all the
other code.

**Reply by Debasish-Mahapatra** [reply]
Hey Will.!
yup, I guess that's what was causing the issue..... Thanks.

**Comment by HathewayWill** [comment]
I would try opening a new terminal

Type contact deactivate to exit (base)

Then try running thr code.

Please note you'll need to download the latest version  1.0.3
